core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -441,9 +442,15 @@ void invokeProxyConnectFailed(SessionProtocol protocol, PoolKey poolKey, Throwab
     }
 
     private static InetSocketAddress toRemoteAddress(PoolKey key) throws UnknownHostException {
```

Comment:
How about moving this method into a member method of `PoolKey` like `poolKey.toRemoteAddress()` for better encapsulation?